java/code/src/com/redhat/rhn/manager/system/entitling/SystemEntitler.java

```diff
@@ -176,4 +191,12 @@ private Optional<ServerGroup> findServerGroupToEntitleAnEntitledServer(Server se
         }
         return serverGroup;
     }
+
+    private void updateLibvirtEngine(MinionServer minion) {
+        Map<String, Object> pillar = new HashMap<>();
+        pillar.put("virt_entitled", minion.hasVirtualizationEntitlement());
+        saltApi.callSync(State.apply(Collections.singletonList("virt.engine-events"),
```

Comment:
I don't see any `virt/engine-events.sls` state in this PR... am I blind? The libvirt engine state was different from the beacon: it threw all libvirt events to the salt events bus in a different way while the beacon gave a full diff every few minutes. I haven't seen any state enabling the beacon in the PR either.

Reply:
Any idea when you dropped this? It was not part of the PR which dropped the virtpoller.

Reply:
>  I haven't seen any state enabling the beacon in the PR either.

This was also not part of the restored commit. Any idea where this was implemented?

Reply:
The beacon configuration seems to happen via pillar. We write the pillar, but the question is, what needs to be done to bring this to the client. I think when you change the virtualization entitlement a highstate should be executed.